packages/editor/src/store/reducer.js

```diff
@@ -264,7 +281,7 @@ export const editor = flow( [
 					( key ) => getPostRawValue( action.post[ key ] );
 
 				return reduce( state, ( result, value, key ) => {
-					if ( value !== getCanonicalValue( key ) ) {
+					if ( ! isEqual( value, getCanonicalValue( key ) ) ) {
```

Comment:
Granting there's one odd edge case for which this doesn't cover: If updates are applied, individual keys won't be unset unless _all_ meta values are now equal with their new canonical values. e.g.

```
{ type: 'EDIT_POST', edits: { meta: { a: 1, b: 2 } } }
// { edits: { meta: { a: 1, b: 2 } } }

{ type: 'UPDATE_POST', edits: { meta: { a: 1 } } }
// { edits: { meta: { a: 1, b: 2 } } }

{ type: 'UPDATE_POST', edits: { meta: { a: 1, b: 2 } } }
// { edits: {} }
```

In practice, I don't think this would ever occur (`UPDATE_POST` is called on save with all edited values assumed as being the new canonical), and the impact being that it would continue to save _edited_ values which had already been saved before.

The alternative would be an implementation which would individually unset keys and remove the top-level key once empty.